### PR TITLE
util/threadpool: fix test_task_context

### PR DESCRIPTION
**Pull request by overvenus**
The rx receives out of order, because it's an asynchronous channel.

### CODE REVIEW

**Comment by BusyJay** [comment]
LGTM

**Comment by choleraehyq** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/util/threadpool.rs [hunk 1]
```diff
@@ -389,19 +386,19 @@ mod test {
...
         for _ in 0..10 {
             task_pool.execute(move |_: &mut TestContext| {});
         }
-        let mut fin: isize = -1;
         for _ in 0..10 {
-            fin = rx.recv_timeout(Duration::from_millis(20)).unwrap();
+            rx.recv_timeout(Duration::from_millis(20)).unwrap();
         }
+        // Check out counter ASAP, since on_tick may be called even if there is no task.
+        assert_eq!(ctx.counter.load(Ordering::SeqCst), 10);
```
Comment (BusyJay) [inline src/util/threadpool.rs]: Why not use `>=`?

File: src/util/threadpool.rs [hunk 2]
```diff
@@ -397,8 +397,8 @@ mod test {
         for _ in 0..10 {
             rx.recv_timeout(Duration::from_millis(20)).unwrap();
         }
-        // Check out counter ASAP, since on_tick may be called even if there is no task.
-        assert_eq!(ctx.counter.load(Ordering::SeqCst), 10);
+        // Check out the counter ASAP, since on_tick may be called even if there is no task.
```
Comment (BusyJay) [inline src/util/threadpool.rs]: The comment is stale. It doesn't have to be checked ASAP.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**